src/formatting/newline_style.rs

```diff
@@ -0,0 +1,181 @@
+use crate::NewlineStyle;
+
+/// Apply this newline style to the formatted text. When the style is set
+/// to `Auto`, the `raw_input_text` is used to detect the existing line
+/// endings.
+///
+/// If the style is set to `Auto` and `raw_input_text` contains no
+/// newlines, the `Native` style will be used.
+pub(crate) fn apply_newline_style(
+    newline_style: NewlineStyle,
+    formatted_text: &mut String,
+    raw_input_text: &str,
+) {
+    match effective_newline_style(newline_style, raw_input_text) {
+        EffectiveNewlineStyle::Windows => {
+            *formatted_text = convert_to_windows_newlines(formatted_text);
+        }
+        EffectiveNewlineStyle::Unix => {}
```

Comment:
Could you add a conversion to unix newlines ? It may be relevant in case where the file got edited on a windows environment first.

Reply:
Implemented in f54fc2f